bgpd/bgp_main.c

```diff
@@ -188,7 +188,8 @@ static __attribute__((__noreturn__)) void bgp_exit(int status)
 			continue;
 		bgp_delete(bgp);
 	}
-	bgp_delete(bgp_default);
+	if (bgp_default)
+		bgp_delete(bgp_default);
```

Comment:
Isn't it better to add the `NULL` check inside the `bgp_delete()` function? This way we also avoid this problem in places we weren't looking.

Reply:
As discussed during the tech call today: if this is supposed to be the only call where `bgp_default` might be default, then there should be no harm to `assert()` that parameters passed to `bgp_delete` are always not `NULL`.

Reply:
I did?

Reply:
I did not push it up.  Got interrupted by a phone call.  Pushing it up here shortly